### PR TITLE
Restructure 2-stream shortwave solver

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Restructure 2-stream shortwave solver

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/rte/shortwave1scalar.jl [hunk 1]
```diff
@@ -1,78 +1,136 @@
...
-                ClimaComms.@threaded device for gcol in 1:ncol
-                    igpt == 1 && set_flux_to_zero!(flux_sw, gcol)
-                    compute_optical_props!(op, as, gcol, igpt, lookup_sw, lookup_sw_cld)
-                    rte_sw_noscat_solve_kernel!(flux, op, bcs_sw, igpt, solar_src_scaled, gcol, nlev)
-                    n_gpt > 1 && add_to_flux!(flux_sw, flux, gcol)
-                end
+    @inbounds begin
+        for igpt in 1:n_gpt
+            ClimaComms.@threaded device for gcol in 1:ncol
+                igpt == 1 && set_flux_to_zero!(flux_sw, gcol)
```
Comment (szy21) [inline src/rte/shortwave1scalar.jl:81]: Do we still need this line?

File: src/rte/shortwave2stream.jl [hunk 2]
```diff
@@ -1,81 +1,129 @@
...
+    @inbounds begin
+        bld_cld_mask = as.cld_mask_type isa AbstractCloudMask
+        for igpt in 1:n_gpt
+            ClimaComms.@threaded device for gcol in 1:ncol
+                bld_cld_mask && Optics.build_cloud_mask!(
+                    view(as.cld_mask_sw, :, gcol),
+                    view(as.cld_frac, :, gcol),
+                    as.cld_mask_type,
+                )
+                igpt == 1 && set_flux_to_zero!(flux_sw, gcol)
```
Comment (szy21) [inline src/rte/shortwave2stream.jl:91]: Do we still need this line?

**Review by szy21 — APPROVED** [review]
The new code structure looks good to me, thanks! We do need `flux_dn_dir`, e.g. for the land.

File: src/rte/shortwave1scalar.jl (hunk 1 above)
Comment (sriharshakandala) [reply src/rte/shortwave1scalar.jl:81]: Yes. It is needed for non-gray/muliti-g-point simulations!

File: src/rte/shortwave2stream.jl (hunk 2 above)
Comment (sriharshakandala) [reply src/rte/shortwave2stream.jl:91]: Yes. It is needed for non-gray/muliti-g-point simulations!

**Comment by sriharshakandala** [comment]
> The new code structure looks good to me, thanks! We do need `flux_dn_dir`, e.g. for the land.

If we just need the `flux_dn_dir` at the surface, we can further reduce memory footprint! 
I can take care of that in a different PR!

**Comment by szy21** [comment]
> > The new code structure looks good to me, thanks! We do need `flux_dn_dir`, e.g. for the land.
> 
> If we just need the `flux_dn_dir` at the surface, we can further reduce memory footprint! I can take care of that in a different PR!

Sounds good, thanks!

**Pull request merged**